aiida/cmdline/utils/migration/v03_to_v04.py

```diff
@@ -0,0 +1,528 @@
+            if 'name' in content:
+                content['label'] = content.pop('name')
+            if 'type' in content:
+                content['type_string'] = content.pop('type')
+    # For metadata.json
+    metadata_group = metadata['all_fields_info']['Group']
+    if 'name' in metadata_group:
+        metadata_group['label'] = metadata_group.pop('name')
+    if 'type' in metadata_group:
+        metadata_group['type_string'] = metadata_group.pop('type')
+
+
+def migration_dbgroup_type_string_change_content(data):
+    """Apply migrations: 0022 - REV. 1.0.22
+    Change type_string according to the following rule:
+    '' -> 'user'
+    'data.upf.family' -> 'data.upf'
+    'aiida.import' -> 'auto.import'
+    'autogroup.run' -> 'auto.run'
+    """
+    for content in data['export_data']['Group'].values():
+        if content['type_string'] == '':
+            content['type_string'] = 'user'
+        elif content['type_string'] == 'data.upf.family':
+            content['type_string'] = 'data.upf'
+        elif content['type_string'] == 'aiida.import':
+            content['type_string'] = 'auto.import'
+        elif content['type_string'] == 'autogroup.run':
+            content['type_string'] = 'auto.run'
+
+
+def migration_calc_job_option_attribute_keys(data):
+    """Apply migrations: 0023 - REV. 1.0.23
+    `custom_environment_variables` -> `environment_variables`
+    `jobresource_params` -> `resources`
+    `_process_label` -> `process_label`
+    `parser` -> `parser_name`
+    """
+
+    # Helper function
+    def _migration_calc_job_option_attribute_keys(attr_id, content):
+        """Apply migration 0023 - REV. 1.0.23 for both `node_attributes*` dicts in `data.json`"""
+        if data['export_data']['Node'][attr_id]['type'] == 'node.process.calculation.calcjob.CalcJobNode.':
+            for key in content:
+                # custom_environment_variables -> environment_variables
+                if key == 'custom_environment_variables':
+                    content['environment_variables'] = content.pop('custom_environment_variables')
+                elif key.startswith(r'custom\_environment\_variables'):
+                    new_key = key.replace(r'custom\_environment\_variables', 'environment_variables', 1)
+                    content[new_key] = content.pop(key)
+
+                # jobresource_params -> resources
+                elif key == 'jobresource_params':
+                    content['resources'] = content.pop('jobresource_params')
+                elif key.startswith(r'jobresource\_params.'):
+                    new_key = key.replace(r'jobresource\_params', 'resources', 1)
+                    content[new_key] = content.pop(key)
+
+                # parser -> parser_name
+                elif key == 'parser':
+                    content['parser_name'] = content.pop('parser')
+
+        # _process_label -> process_label
+        elif data['export_data']['Node'][attr_id]['type'].startswith('node.process.'):
```

Comment:
this should be a normal `if` because it is now not hit for `CalcJobNodes`

Reply:
done